docker/Dockerfile

```diff
@@ -15,30 +66,23 @@ RUN yum update -y \
         wget \
         aws-cli \
         git \
-    && rm -rf /var/lib/apt/lists/*
-
-RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
+    && yum clean all \
+    && rm -rf /var/cache/yum
 
-# Base image has an old aws cli version (v1.x), this will replace that
+# Upgrade AWS CLI to v2
 ADD https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip /tmp
 RUN unzip /tmp/awscli-exe-linux-x86_64.zip -d /tmp \
     && rm /tmp/awscli-exe-linux-x86_64.zip \
     && /tmp/aws/install
 
-COPY . /asset-input
-
-RUN --mount=type=ssh cd /asset-input \
-    && python3 -m pip install --upgrade pip \
-    && pip3 install .[lambda] \
-        --no-cache-dir \
-        --target "${LAMBDA_TASK_ROOT}" \
-    && cp docker/docker-entrypoint.sh ${LAMBDA_TASK_ROOT}/docker-entrypoint.sh \
-    && chmod +x ${LAMBDA_TASK_ROOT}/docker-entrypoint.sh
+# Copy only the vendored application deps into the Lambda task root
+# (No uv, no build tools end up in the final image)
+COPY --from=builder /opt/lambda-task ${LAMBDA_TASK_ROOT}
 
 ENV PYTHONPATH="${LAMBDA_TASK_ROOT}:${PYTHONPATH}"
 ENV PATH="${LAMBDA_TASK_ROOT}/bin:${PATH}"
 
-RUN rm -rf /asset-input
+# Clean up build-time artifacts that might have been copied (defensive)
+RUN find ${LAMBDA_TASK_ROOT} -type d -name "__pycache__" -prune -exec rm -rf {} +
```

Comment:
We should be doing this cleanup step in the `stage 1` in the builder, not here. Because the act of `COPY --from=builder /opt/lambda-task ${LAMBDA_TASK_ROOT}` on line 80 already created an image layer in that persists even if you a separate `RUN` cleanup step here.